sycl/CMakeLists.txt

```diff
@@ -22,10 +22,27 @@ if (NOT DEFINED SYCL_ENABLE_PLUGINS)
   set(SYCL_ENABLE_PLUGINS "opencl;level_zero")
 endif()
```

Comment:
Certainly something we could do. SYCL uses the term backend as well which seems to have a 1:1 relationship with plugin/adapter. Should this follow the SYCL backend naming convention instead? e.g. `SYCL_ENABLE_BACKENDS` might make more sense to a SYCL user.

Reply:
I think that compiler sources are more about developers, but I like the idea of using `backend` naming instead, because it immediately connects this variable to a well-defined term in the SYCL spec, i.e. it should be clear what a side effect of changing this will be.

Considering that this is just one PR in a series, we can reserve this for later

Reply:
captured in https://github.com/intel/llvm/issues/14598 for future work